### ISSUE TITLE
Frontend: css, js, jade building. bower

### ISSUE THREAD

**Issue by boxfrommars** [Rutorika/laravel-rules#8]
тут первоочередная задача определить, будем ли мы использовать [elixir](http://laravel.com/docs/5.1/elixir) (laravel-обёртка над gulp) или чистый [gulp](http://gulpjs.com/)

**плюсы elixir:**
- лаконичность, он уже сконфигурирован для работы с структурой папок laravel.
- связь с laravel, я нашёл проявление этого в:  в .gulpfile есть доступ к значению окружения (production/local), автоматическая версионность подключённых скриптов на странице. пока немного, но нужно ещё изучить
- ~~я нашёл [плагин для jade](https://github.com/CREEATION/laravel-elixir-jade) который поддерживает blade-синтаксис~~

**минусы elixir**
- лаконичность, которая не конфигурируется, например, я так и не нашёл как изменить структуру директории `resource`, чтобы разделить её на админскую и фронтенд часть
- в любом случае это дополнительная обёртка над тем, что и так справляется со своими задачами, причём эта обёртка заранее сконфигурирована (и тяжела для переконфигурации)
- уход от gulp-community, теперь приходится искать не gulp-плагин (коих уже миллионы), а elixir-плагин, который в конечном итоге -- обёртка над gulp-плагином. то есть опять же, это добавление ещё одного уровня сложности (причём не разработчиком оригинального gulp-плагина, а, скорее всего, разработчиком на laravel/php)

также нужно решить насчёт [bower](http://bower.io/). будем ли мы его использовать для подключения сторонних клиентских библиотек.
**плюсы**
- удобно подключать и обновлять пакеты и их зависимости

**минусы**
- дополнительная технология
- скорее всего не получится использовать разные версии библиотек для фронтед- и админ-частей
### вопросы для обсуждения:
- будем ли использовать elixir
- будем ли использовать bower

в любом случае, нам нужно максимально аккуратно и понятно как для frontend- так и для bakend-разработчиков решить следующие задачи:
- сборка и минификация js и css файлов как для фронтенд, так и для админки
- компиляция jade-темплейтов в blade-темплейты (только фронтенд?)
- оптимизация картинок

**Comment by boxfrommars** [comment]
у меня получилось с помощью чистого gulp сделать билд jade-темплейтов в blade-темплейты со всеми blade-конструкциями: https://github.com/boxfrommars/laravel-jade-test

есть несколько замечаний:
- миксины надо подключать во все файлы (я никогда не работал с jade, может это и всем известно :)
- мы не пользуемся инклудами jade только блейд-возможностями +extends +yield +include
- подробности синтаксиса для управляющих конструкций blade см. в [файле миксина](https://github.com/boxfrommars/laravel-jade-test/blob/master/resources/jade-views/mixins/blade-mixin.jade)
- папка resources/views/frontend/ полностью очищается при билде, туда ничего нельзя класть руками, только скомпилированные шаблоны jade

**Comment by boxfrommars** [comment]
#### будем ли использовать elixir

нет не будем

#### будем ли использовать bower

да, но опционально

#### сборка и минификация js и css файлов как для фронтенд, так и для админки

[gulp](http://gulpjs.com/), подробности по структуре публичной директории и расположению билдов см. в #2

[пример gulp-файла](https://gist.github.com/boxfrommars/de80dbeb59145d6bcb00) для структуры описанной в #2 

#### компиляция jade-темплейтов в blade-темплейты

полностью на совести верстальщика

#### оптимизация картинок

статичные картинки оптимизируем с помощью gulp (например [gulp-imagemin](https://github.com/sindresorhus/gulp-imagemin)). загружаемые пользователями картинки или не оптимизируем (на совести загружающего). или если сильно заставляют, то [laravel-image-optimizer](https://github.com/approached/laravel-image-optimizer), но даём понять, что это зло и ад

**Comment by alisabeata** [comment]
хотела высказаться на счет именования папок в структуре. Если заменить css на styles, js > scripts, img > images, было бы лучше. Я за использование более абстрактных названий, хотя, это в большей степени относится к css здесь, остальное, чтобы выдержать стиль)

**Comment by DexinDev** [comment]
со стилями - согласен, а со скриптами - нет. Ведь там именно js будет, не perl, php или ruby.
img или images пофигу.

**Comment by boxfrommars** [comment]
@alisa-beata @DexinDev да, мне не нравится примерно поэтому же. если ты пишешь на кофескрипте и sass, например, то лучше будет создать в ассетах папки `sass` и `coffee` (не знаю, какое там правильное сокращение), иначе это предполагает, что в папке styles может быть и css, и sass, и всё это вперемешку. насчёт img -- это на мой взгляд стандарт

**Comment by DexinDev** [comment]
@boxfrommars тогда ок. Ассеты с кодом называем по расширениям
